lib/presentation/view/upcoming_movies_view.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:flutter/material.dart';
+import '../../core/util/constants/ui_constants.dart';
+import '../../core/util/enums/endpoint.dart';
+import '../../data/model/data_state.dart';
+import '../../domain/entity/movie.dart';
+import '../bloc/movies_bloc.dart';
+import '../widget/custom_drawer.dart';
+import 'movie_details_view.dart';
+
+class UpcomingMoviesView extends StatefulWidget {
+  const UpcomingMoviesView({
+    super.key,
+    required this.bloc,
+  });
+
+  final MoviesBloc bloc;
+  @override
+  State<UpcomingMoviesView> createState() => _UpcomingMoviesViewState();
+}
+
+class _UpcomingMoviesViewState extends State<UpcomingMoviesView> {
+  static const String errorMessage = 'An error has occurred while loading data';
+  @override
+  void initState() {
+    super.initState();
+    widget.bloc.getMovies(Endpoint.upcoming);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final PageController controller = PageController();
+    return Scaffold(
+      backgroundColor: Colors.black12,
+      appBar: AppBar(
+        title: const Center(
+          child: Text(
+            MovieDetailsUiConstants.upcomingMoviesLabel,
+          ),
+        ),
+      ),
+      body: StreamBuilder<DataState>(
+          initialData: widget.bloc.initialData,
+          stream: widget.bloc.movies,
+          builder: (context, snapshot) {
```

Comment:
Add types of parameters and also review the DataState class

Reply:
done